discord/app_commands/transformers.py

```diff
@@ -679,4 +714,12 @@ def annotation_to_parameter(annotation: Any, parameter: inspect.Parameter) -> Co
     if parameter.kind in (parameter.POSITIONAL_ONLY, parameter.VAR_KEYWORD, parameter.VAR_POSITIONAL):
         raise TypeError(f'unsupported parameter kind in callback: {parameter.kind!s}')
 
+    if not hasattr(inner.autocomplete.__func__, '__transformer_autocomplete_not_overridden__'):
+        from .commands import _validate_auto_complete_callback
+
+        async def func(*args, **kwargs):
+            return await inner.autocomplete(*args, **kwargs)
+
+        result.autocomplete = _validate_auto_complete_callback(func, skip_binding=True)
```

Comment:
This doesn't seem proper. This validation doesn't do anything.

Reply:
Yeah I'm not sure how to avoid it though, since no matter what it's going to try setting `.requires_binding` on the callback passed, which unless we always take `__func__` out, is going to fail.

Could rework it such that only if requires_binding is True do we set the attribute, and use getattr when checking it?

Reply:
```py
defined_autocomplete = inner.__dict__['autocomplete']
if isinstance(defined_autocomplete, (classmethod, staticmethod)):
    result.autocomplete = _validate_auto_complete_callback(inner.autocomplete, skip_binding=True)
else:
    raise TypeError('autocomplete method must be a classmethod or a staticmethod')
```